Flaky: TestAllocatorCrossNamespace

Flakiness:

```
go test -mod=vendor -race -parallel=64 agones.dev/agones/test/e2e --kubeconfig /root/.kube/config \
	--gameserver-image=gcr.io/agones-images/udp-server:0.21 \
	--pullsecret= \
	--feature-gates=""
time="2020-06-02 00:21:51.232" level=info msg="Starting e2e test(s)" featureGates="ContainerPortAllocation=false&Example=true&PlayerTracking=false" gameServerImage="gcr.io/agones-images/udp-server:0.21" perfOutputDir= pullSecret= stressTestLevel=0 version=
time="2020-06-02 00:21:51.233" level=info msg="Cleaning up now."
time="2020-06-02 00:21:51.377" level=info msg="Finished cleanup."
time="2020-06-02 00:21:54.854" level=info msg="waiting for fleet condition" fleet=simple-fleet-69f47
time="2020-06-02 00:21:54.883" level=info msg="fleet simple-fleet-69f47 has 0/3 ready replicas"
time="2020-06-02 00:21:56.889" level=info msg="fleet simple-fleet-69f47 has 0/3 ready replicas"
time="2020-06-02 00:21:58.895" level=info msg="fleet simple-fleet-69f47 has 0/3 ready replicas"
time="2020-06-02 00:22:00.889" level=info msg="fleet simple-fleet-69f47 has 3/3 ready replicas"
time="2020-06-02 00:22:00.927" level=info msg="failing Allocate request" error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 35.233.192.59:443: connect: connection refused\""
time="2020-06-02 00:22:02.938" level=info msg="failing Allocate request" error="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: Error while dialing dial tcp 35.233.192.59:443: connect: connection refused\""
--- FAIL: TestAllocatorCrossNamespace (2.94s)
    allocator_test.go:116: Allocator TLS is refreshed with public CA: -----BEGIN CERTIFICATE-----
        MIIDMjCCAhqgAwIBAgIRAPft9kUnNzMF01EFw657+nUwDQYJKoZIhvcNAQEFBQAw
        KjEQMA4GA1UEChMHdGVzdGluZzEWMBQGA1UEAxMNMzUuMjMzLjE5Mi41OTAeFw0y
        MDA2MDIwMDIyMDVaFw0yMDA2MDIwMTIyMDVaMCoxEDAOBgNVBAoTB3Rlc3Rpbmcx
        FjAUBgNVBAMTDTM1LjIzMy4xOTIuNTkwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAw
        ggEKAoIBAQDqSwAYNischTPwnWMo94/mXftoe+PYYXerPzxYZ4KtA5aMxwyAb+XG
        U0LDA1fYUAYoiEltiYbJppY4gEGduEG4Cs1CuHCU5gT/ACG553G3MrqZNU9xxEX2
        JHZQzbFwEi/vI7JnMIWeXcWXsVWVrGgyEEHxww7LrLtdDOex/mCATQatbEsoo722
        rGnRo64NaZ3OcY/pmqmRvGimPI6jtyiH7edErqaR4+08paYhmBhvIPdaw1eCWijt
        iuxQlr6IPmprsOAPei1MVEc9c9yaPPgPNXlFKrHJt1EV1tpWdNP7IooH1VybSAS+
        Qb7PQyIz22IQZ/PQfqpfKFRpxUlWBMgbAgMBAAGjUzBRMA4GA1UdDwEB/wQEAwIC
        pDAdBgNVHSUEFjAUBggrBgEFBQcDAgYIKwYBBQUHAwEwDwYDVR0TAQH/BAUwAwEB
        /zAPBgNVHREECDAGhwQj6cA7MA0GCSqGSIb3DQEBBQUAA4IBAQAaUSXWG6PkpDbi
        go4tiSdtO8Z3oSLzfLWJ7ob7nx8y1MiLdQ+K6sihu3VZAzh3lvo7pSWJ8kP6poXc
        MOjym3HI6/3v15y2zHQRLv5g1DxFWgzUkUJno22EH9kPxvjWUaDJ/Q0I3FWVs75r
        dmw6EqSxPRXB7RTi8vGbnT4wwdKrBCqjvRxg3VNJMTeIYkKyPsaqcusfwcn7xJwb
        gbtR+XoL57Ygr7YbcBWv8REnR9Taz16sx4HF7XVOV2xsd9/7RPHhlxKa+ZM04pIJ
        A57F/113tYHnqWjxXt5jOPNZJtG/4l+KbjA4mTQLn7jlMW3G4Dosu9nxraaSSQIR
        E0FV51R1
        -----END CERTIFICATE-----
         for endpoint 35.233.192.59
    allocator_test.go:117: deleting pods failed: pods "agones-allocator-79bdd9866d-skjr7" not found
```

Looks like it might be an issue with race conditions in the cleanup code. We should probably ignore not found errors.

--- a/test/e2e/allocator_test.go
+++ b/test/e2e/allocator_test.go
@@ -39,6 +39,7 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/wait"
@@ -273,6 +274,10 @@ func restartAllocator(t *testing.T) {
 			continue
 		}
 		if err := kubeCore.Pods(agonesSystemNamespace).Delete(pod.Name, &metav1.DeleteOptions{}); err != nil {
+			if k8serrors.IsNotFound(err) {
+				logrus.WithField("pod", pod.Name).WithError(err).Warn("Attempt at deletion failed")
+				continue
+			}
 			t.Fatalf("deleting pods failed: %s", err)
 		}
 	}

